### PR TITLE
feat: Add example Go stack

### PR DESCRIPTION
**Pull request by lym953**
### What does this PR do?
Adds an example Go CDK stack to demonstrate how to integrate Datadog with CDK.

I did this by:
1. create a demo CDK project following [Tutorial: Create your first AWS CDK app](https://docs.aws.amazon.com/cdk/v2/guide/hello_world.html)
2. add Datadog-related setup in `app.go`
3. modify `README.md`. Most of the content was copied from other example stacks.

#### Steps
1. run `go get` to get dependencies
2. run `go build` to build the binary file `app`
3. run `aws-vault exec sso-serverless-sandbox-account-admin -- cdk bootstrap --toolkit-stack-name yiming-go-cdk-test --qualifier yiming` to bootstrap the environment (optional)
4. run `aws-vault exec sso-serverless-sandbox-account-admin -- cdk deploy` to deploy the stack
5. in AWS UI, run the Lambda function a few times

#### Result
The Lambda function showed up on Datadog UI
image

### Additional Notes

Next steps:
1. Delete the folder `dist/go/ddcdkconstruct` from git since that has been moved to a separate repo https://github.com/DataDog/datadog-cdk-constructs-go
2. Update `README.md` by adding instructions for Go
3. Clean up unused files on the root level of `datadog-cdk-constructs-go` repo
4. Add the Go release script to `publish_prod.sh` to make release more automated
5. Maybe release a new version

[Feature Request: Datadog CDK Construct for Go](https://datadoghq.atlassian.net/browse/FRSLES-665)

### Types of Changes

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [x] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog

### CODE REVIEW

File: examples/go-stack/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Datadog CDK Go Example
```
Comment (lym953) [inline examples/go-stack/README.md:1]: This file is modified from `python-stack/README.md`

File: examples/python-stack/README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Datadog CDK Python Example
 
-Use this example Python stack to try out the [datadog-cdk-constructs](https://github.com/DataDog/datadog-cdk-constructs) v2 library. It contains Node, Python and Go Lambda functions.
+Use this example Python stack to try out the [datadog-cdk-constructs](https://github.com/DataDog/datadog-cdk-constructs) v2 library. It contains Node, Python, and Go Lambda functions.
```
Comment (lym953) [inline examples/python-stack/README.md:3]: To address documentation team's comments on https://github.com/DataDog/datadog-cdk-constructs/pull/271

File: examples/go-stack/.gitignore [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Binaries for programs and plugins
+*.exe
+*.exe~
+*.dll
+*.so
+*.dylib
+app
```
Comment (lym953) [inline examples/go-stack/.gitignore:7]: This line was added by me. Other lines are all generated.

File: examples/go-stack/app.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+			FlushMetricsToLogs:    jsii.Bool(true),
+			Site:                  jsii.String("datadoghq.com"),
+			ApiKey:                jsii.String(os.Getenv("DD_API_KEY")),
+			EnableDatadogTracing:  jsii.Bool(true),
+			EnableMergeXrayTraces: jsii.Bool(true),
+			EnableDatadogLogs:     jsii.Bool(true),
+			InjectLogContext:      jsii.Bool(true),
+			LogLevel:              jsii.String("debug"),
+		})
+	datadog.AddLambdaFunctions(&[]interface{}{myFunction}, nil)
```
Comment (lym953) [inline examples/go-stack/app.go:67]: Manual code

File: examples/go-stack/app.go [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+	app.Synth(nil)
+}
+
+// env determines the AWS environment (account+region) in which our stack is to
+// be deployed. For more information see: https://docs.aws.amazon.com/cdk/latest/guide/environments.html
+func env() *awscdk.Environment {
+	env := awscdk.Environment{
+		Account: jsii.String("425362996713"),
+		Region:  jsii.String("us-east-1"),
+	}
```
Comment (lym953) [inline examples/go-stack/app.go:92]: Manual code.

File: examples/go-stack/cdk.json [hunk 6]
```diff
@@ -0,0 +1,68 @@
+{
```
Comment (lym953) [inline examples/go-stack/cdk.json:1]: This file is generated.

File: examples/go-stack/go.mod [hunk 7]
```diff
@@ -0,0 +1,26 @@
+module app
```
Comment (lym953) [inline examples/go-stack/go.mod:1]: This file is generated from running `go get`

File: examples/go-stack/go.sum [hunk 8]
```diff
@@ -0,0 +1,48 @@
+github.com/DataDog/datadog-cdk-constructs-go/ddcdkconstruct v1.13.0 h1:AOgs4MEGLb0gmkEdBx2HC7BrUtrpM46tyET+cJ7t80M=
```
Comment (lym953) [inline examples/go-stack/go.sum:1]: This file is generated from running `go get`

File: README.md [hunk 9]
```diff
@@ -16,7 +16,7 @@ This CDK library automatically configures ingestion of metrics, traces, and logs
 
 ## AWS CDK v1 vs AWS CDK v2
 
-**WARNING**:  `AWS CDK v1` has reached end-of-support and `datadog-cdk-constructs` will no longer be receiving updates. It's strongly recommended to upgrade to `AWS CDK v2` ([official migration guide](https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html)) and switch to using `datadog-cdk-constructs-v2`.
+**WARNING**: `AWS CDK v1` has reached end-of-support and `datadog-cdk-constructs` will no longer be receiving updates. It's strongly recommended to upgrade to `AWS CDK v2` ([official migration guide](https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html)) and switch to using `datadog-cdk-constructs-v2`.
```
Comment (lym953) [inline README.md:19]: Lots of changes on this file was formatting changes automatically done by VS Code. Let me you if you think I should revert them.

File: README.md [hunk 10]
```diff
@@ -97,67 +113,93 @@ datadog.addLambdaFunctions([<LAMBDA_FUNCTIONS>])
...
+    datadog.AddLambdaFunctions(&[]interface{}{myFunction}, nil)
+    datadog.AddForwarderToNonLambdaLogGroups()
+```
+
 ## Source Code Integration
+
 [Source code integration](https://docs.datadoghq.com/integrations/guide/source-code-integration/) is enabled by default through automatic lambda tagging, and will work if:
 
-- The Datadog Github Integration is installed.
+- The Datadog Github integration is installed.
```
Comment (lym953) [inline README.md:140]: Addressing the comment from documentation team
https://github.com/DataDog/datadog-cdk-constructs/pull/271#discussion_r1702365958

File: README.md [hunk 11]
```diff
@@ -97,67 +113,93 @@ datadog.addLambdaFunctions([<LAMBDA_FUNCTIONS>])
...
 **Note: these alternate guides only work for Typescript.**
+
 <details>
-  <summary>datadog-cdk version satisfied, but Datadog Github Integration NOT installed</summary>
+  <summary>datadog-cdk version satisfied, but Datadog Github integration NOT installed</summary>
 
-  If the Datadog Github Integration is not installed, you need to import the `datadog-ci` package and manually upload your Git metadata to Datadog.
-  We recommend you do this where your CDK Stack is initialized.
+If the Datadog Github integration is not installed, you need to import the `datadog-ci` package and manually upload your Git metadata to Datadog.
+For the best results, import the `datadog-ci` package where your CDK Stack is initialized.
```
Comment (lym953) [inline README.md:155]: Addressing the comment from documentation team
https://github.com/DataDog/datadog-cdk-constructs/pull/271/files/6a32a452d56f58ff6f6cf936ff9d7bd0449fe02c#r1702366653

**Review by TalUsvyatsky — APPROVED** [review]

File: README.md [hunk 12]
```diff
@@ -97,67 +113,93 @@ datadog.addLambdaFunctions([<LAMBDA_FUNCTIONS>])
...
+```
 
-  app.synth();
-  ```
 </details>
 <details>
   <summary>datadog-cdk version NOT satisfied</summary>
 
-  Change your initialization function as follows (note: we're changing this to pass just the `gitHash` value to the CDK):
+Change your initialization function as follows (note: we're changing this to pass just the `gitHash` value to the CDK):
```
Comment (urseberry) [inline README.md:175]:
```suggestion
Change your initialization function as follows (**Note:** We're changing this to pass just the `gitHash` value to the CDK):
```
Can you rephrase the note to not use "we?" I'm not sure if replacing it with "Datadog" is appropriate here, so I didn't make a wording suggestion.

**Review by urseberry — APPROVED** [review]
Left some nonblocking feedback.

Thank you for linking to my previous review! It saved the documentation team from repeating the work.

File: README.md (hunk 12 above)
Comment (lym953) [reply README.md:175]: I agree that "we" is confusing. Let me change it to:
Change your initialization function as follows (in this case, `gitHash` value is passed to the CDK):
